modules/gapi/src/backends/ie/giebackend/giewrapper.hpp

```diff
@@ -51,11 +52,27 @@ GAPI_EXPORTS IE::Core getPlugin(const GIEParam& params);
 GAPI_EXPORTS inline IE::ExecutableNetwork loadNetwork(      IE::Core&       core,
                                                       const IE::CNNNetwork& net,
                                                       const GIEParam& params) {
-    return core.LoadNetwork(net, params.device_id);
+    if (params.rctx != nullptr) {
+        return core.LoadNetwork(net, params.rctx);
+    } else {
+        return core.LoadNetwork(net, params.device_id);
+    }
 }
 GAPI_EXPORTS inline IE::ExecutableNetwork importNetwork(      IE::Core& core,
-                                                        const GIEParam& param) {
-    return core.ImportNetwork(param.model_path, param.device_id, {});
+                                                        const GIEParam& params) {
+    if (params.rctx != nullptr) {
+        GAPI_Assert(params.model_path.find(".blob") != std::string::npos);
```

Comment:
why do we need this assert here? Is it required only here and not in the below case?

Reply:
Removed the assert. Anyway, I believe it is the only possible way to pass it here